servers/rendering/rendering_server_default.cpp

```diff
@@ -368,8 +370,10 @@ Size2i RenderingServerDefault::get_maximum_viewport_size() const {
 void RenderingServerDefault::_assign_mt_ids(WorkerThreadPool::TaskID p_pump_task_id) {
 	server_thread = Thread::get_caller_id();
 	server_task_id = p_pump_task_id;
+#ifdef RD_ENABLED
 	// This is needed because the main RD is created on the main thread.
 	RenderingDevice::get_singleton()->make_current();
+#endif // RD_ENABLED
```

Comment:
That's also a hack we should solve somehow, this breaks encapsulation.

CC @godotengine/rendering